.pre-commit-config.yaml

```diff
@@ -14,7 +14,7 @@ repos:
         files: \.py$
```

Comment:
You probably don't want to update this file and push, right @peterjc ?

Reply:
We update the pre-commit setup periodically, I see no reason to touch it on this PR.

Reply:
The modification was not intentional, I reverted back to the original. Also is there anything else I need to do with this pull request?

Reply:
Try to avoid doing any merges - rather if you know how to and are comfortable doing so, fetch the latest main repository, and rebase onto our master branch.

(Biopython tried to keep a simple linear history, so any PR will either be done as a rebase, or a squash-commit which reduces all your commits into one]

Reply:
Sadly I am not very familiar. Am I correct for assuming I can leave this as is for now, and if the PR gets merged it will be overwritten?

Reply:
It should be fine - one of us can do a manual rebase if required.

Reply:
Perfect, thanks a lot! Is there anything in general I need  to do, or the request is good as is?

Reply:
Nothing in terms of rebasing or merges. See the comments from Joao though, most have not been resolved so I assume need addressing still.

Reply:
I think I resolved all of the issues. I reconstructed the code as requested and also finished the minor things that were brought up

Reply:
I think I resolved all the things that were asked. I reconstructed the whole code and also implemented the minor things that were asked